dlp/snippets/src/main/java/dlp/snippets/DataProfilePubSubMessageParser.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dlp.snippets;
+
+// [START dlp_data_profile_pub_sub_message_parser]
+
+import com.google.api.core.ApiService;
+import com.google.cloud.pubsub.v1.AckReplyConsumer;
+import com.google.cloud.pubsub.v1.MessageReceiver;
+import com.google.cloud.pubsub.v1.Subscriber;
+import com.google.privacy.dlp.v2.DataProfilePubSubMessage;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.pubsub.v1.ProjectSubscriptionName;
+import com.google.pubsub.v1.PubsubMessage;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class DataProfilePubSubMessageParser {
+
+  public static void main(String... args) throws Exception {
+    // TODO(developer): Replace these variables before running the sample.
+    // The Google Cloud project id to use as a parent resource.
+    String projectId = "your-project-id";
+    String subscriptionId = "your-pubsub-subscription-id";
+    parseMessage(projectId, subscriptionId);
+  }
+
+  // Parses messages received from a Google Cloud Pub/Sub subscription into a
+  // DataProfilePubSubMessage object.
+  public static void parseMessage(String projectId, String subscriptionId) {
+    int timeoutSeconds = 5;
```

Comment:
5 seconds is a very low timeout for pubsub. The user would need to start this sample, then run a job that publishes to pubsub, wait for that job to complete, in order to see pubsub messages.

I think a more reasonable timeout would be 5 minutes. If the user wants to stop the sample they can always Ctrl+C.